### PR TITLE
fix(codex): omit unset background probe parameter

### LINKED ISSUE #228: fix(gateway): Responses API always sends background:false, breaking strict-whitelist upstreams
## Bug: Responses API always sends `background:false`, breaking strict-whitelist upstreams

When TokenHub proxies a `/v1/responses` request through an OpenAI-compatible provider, it **always injects `background:false` into the upstream request body**, even when the client never sent a `background` field.

Some OpenAI-compatible upstreams (e.g. certain sub2api / third-party gateways) apply a **strict parameter whitelist** on the Responses endpoint and reject any parameter they don't implement. Because they don't special-case "`false` equals the default", they return:

```
HTTP 400  {"error":{"message":"Unsupported parameter: background","type":"invalid_request_error"}}
```

This happens for **any** value (`true` / `false` / `null`) — they reject the *presence* of the parameter.

### Root cause

`ResponsesRequest.MarshalJSON` in `backend/internal/server/types.go`:

```go
setRawJSONField(raw, "background", r.Background, true)   // line ~1064
```

The `present` argument is hard-coded to `true`, so `background` is written to the raw map **even when `r.Background` is the zero value `false`** and the client never provided it.

Note the sibling `r.raw == nil` branch does NOT have this problem:

```go
if r.raw == nil {
    return json.Marshal(requestAlias(r))   // struct tag `background,omitempty` correctly omits false
}
```

So only the raw-passthrough path (the normal one for `/v1/responses`, since `UnmarshalJSON` always populates `raw`) is affected.

### Why this matters

- `background` defaults to `false` in OpenAI's API. Explicitly sending `false` is semantically identical to omitting it.
- Codex's official client **never sends the field at all** (confirmed by packet capture).
- Best practice: a field equal to its default should be omitted on the wire. Sending it needlessly breaks compatibility with strict-whitelist upstreams.

### Suggested fix

Only write `background` when it is actually `true` (the non-default value):

```go
setRawJSONField(raw, "background", r.Background, r.Background)
```

This preserves `background:true` for clients that genuinely request background mode, while omitting the field otherwise — matching both the struct-tag branch and the official client behavior.

Happy to open a PR with this change plus a regression test.

### PR DESCRIPTION
**Pull request by Mr0bean**
## Summary

Codex subscription probes started failing with `Unsupported parameter: background` after Responses request marshaling began forcing `background: false` onto requests where the client never supplied the field. The ChatGPT Codex Responses endpoint rejects the presence of this unsupported parameter.

This change preserves the original field-presence semantics: an omitted `background` field stays omitted, while explicitly supplied `background: false` or `background: true` values remain on the wire. This is the original fix by `chuhao` and differs from #230 by preserving an explicit false value.

## Related Issue

Fixes #228.

Alternative implementation to #230.

## Changes

- Omit `background` when it was absent from the incoming Responses request.
- Preserve explicit `background: false` and `background: true` values.
- Add a Provider-level regression assertion for the default Codex subscription probe.
- Add local Codex compatibility coverage and align its usage assertions with the current public OpenAI-compatible usage object.

## Type of Change

- [x] Bug fix
- [ ] New feature
- [ ] Refactor or maintenance
- [ ] Documentation
- [ ] Deployment or configuration

## Verification

- [x] `cd backend && go test ./... -count=1`
- [x] `cd backend && go vet ./...`
- [x] Relevant Codex, compatibility, and metrics tests passed five consecutive runs.
- [x] `node --test tools/*.test.mjs` (122 tests passed)
- [x] `node tools/check-doc-translations.mjs --base origin/main --head HEAD`
- [x] `node tools/check-ui-translations.mjs --base origin/main --head HEAD`
- [x] `node tools/check-env-contract.mjs`
- [x] `node tools/check-source-lines.mjs`
- [x] `git diff --check`
- [ ] Frontend checks were not run because this change only touches Go request marshaling and backend tests.
- [ ] SDK live smoke tests were not run because they require compatible external provider credentials.

## Compatibility, Security, and Operations

The public Responses request contract remains compatible: explicitly supplied boolean values are preserved, while an absent optional field is no longer synthesized. The change only affects outbound Codex subscription requests that did not provide `background`.

No database, credential format, environment variable, deployment, or user-interface changes are introduced. Rollback is a normal code rollback, but it restores the Codex subscription probe failure for strict upstream endpoints.

## Checklist

- [x] The PR title and body are written in English.
- [x] Tests were added or updated for behavior changes, or the reason they are unnecessary is documented.
- [x] No credentials, local `.env` files, databases, backups, or runtime logs are included.
- [x] Environment variable changes are synchronized across examples, Compose, `start.sh`, and deployment documentation where applicable. No environment variables were changed.
- [x] Shared user-facing behavior is documented consistently in English, Simplified Chinese, and Japanese where applicable. No user-facing documentation or copy changed.
- [x] `data/model-catalog.yaml` remains tracked and catalog changes were reviewed where applicable. The catalog was not changed.
- [x] `git diff --check` passes.

### CODE REVIEW

File: backend/internal/server/types.go [hunk 1]
```diff
@@ -1061,7 +1061,7 @@ func (r ResponsesRequest) MarshalJSON() ([]byte, error) {
 	setRawJSONField(raw, "model", r.Model, r.Model != "")
 	setRawJSONField(raw, "input", r.Input, r.Input != nil)
 	setRawJSONField(raw, "stream", r.Stream, true)
-	setRawJSONField(raw, "background", r.Background, true)
+	setRawJSONField(raw, "background", r.Background, r.Background || raw["background"] != nil)
```
Comment (astaxie) [inline backend/internal/server/types.go]: [P2] Preserve or explicitly reject an explicit null value. `UnmarshalJSON` decodes `background:null` into `Background == false` while retaining `raw["background"]`; this condition therefore rewrites the caller's `null` to `background:false` on the wire. Issue #228 explicitly lists null as a rejected presence, and the surrounding comment describes a transparent raw passthrough. Please either preserve the raw null (or reject it at ingress) and add a table-driven regression for omitted/false/true/null; the current provider-probe assertion only covers the omitted path and the existing `Background:true` tests use `raw == nil`, bypassing this branch.

**Review by astaxie — COMMENTED** [review]
Reviewed the complete current diff. The production change fixes the omitted-background case and all current GitHub checks plus the full backend validation pass. I found one P2 compatibility/test gap inline, so this is a comment-only review.

**Review by astaxie — APPROVED** [review]
Re-reviewed the complete current diff at `ed100a28`. The explicit-null follow-up preserves omitted/false/true/null presence semantics through the actual raw passthrough path, and the focused regression covers all four cases. The provider probe assertion also confirms the original omitted-field failure is fixed. Full backend tests, `go vet`, all 122 repository tests, translation/environment/source-line gates, and `git diff --check` pass; all current GitHub checks are green. No blocking issues remain on this head.
